l10n_ch_fds_upload_sepa/wizard/fds_inherit_sepa_wizard.py

```diff
@@ -0,0 +1,319 @@
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api, exceptions
+import logging
+import base64
+import tempfile
+import shutil
+import pysftp
+import os
+
+_logger = logging.getLogger(__name__)
+
+
+class fds_inherit_sepa_wizard(models.TransientModel):
+    ''' This addons allows you to upload the generate SEPA electronic payment
+        file for Switzerland to your FDS Postfinance.
+
+        This addon will add upload button to the SEPA wizard.
+        This Class inherit from l10n_ch_sepa.wiz_pain_001.
+    '''
+    _inherit = 'wizard.pain001'
+
+    fds_account = fields.Many2one(
+        comodel_name='fds.postfinance.account',
+        string='FDS account',
+        help='select one FDS account or create a FDS account'
+    )
+    fds_account_directory = fields.Many2one(
+        comodel_name='fds.postfinance.files.directory',
+        string='FDS files directory',
+        help='select one upload directory or config your upload directory'
+    )
+    attachment_id = fields.Many2one(
+        comodel_name='ir.attachment',
+        string='File attachment',
+        readonly=True,
+        help='[info] keep one recored of the model attachment'
+    )
+    filename = fields.Char(
+        string='Attachment filename',
+        related='attachment_id.name',
+        readonly=True,
+        help='The name of the attachment file'
+    )
+    payment_order_id = fields.Many2one(
+        comodel_name='payment.order',
+        string='Payment order',
+        readonly=True,
+        help='payment order id'
+    )
+    state = fields.Selection(
+        selection=[('default', 'Default'),
+                   ('finish', 'finish'),
+                   ('upload', 'upload'),
+                   ('confirm', 'confirm')],
+        readonly=True,
+        default='default',
+        help='[Info] keep state of the wizard'
+    )
+
+    ##################################
+    #         Button action          #
+    ##################################
+    @api.multi
+    def generateFile_button(self):
+        ''' create a pain_001 file into wizard and add it as an attachment
+            Called by pressing generate button.
+
+            :returns action: configuration for the next wizard's view
+        '''
+        self.ensure_one()
+        self.create_pain_001()
+        attachment_obj = self.env['ir.attachment']
+        payment_order_id = self.env.context.get('active_id')
+        file = attachment_obj.search([['res_id', '=', payment_order_id]])[0]
+        self.write({'attachment_id': file.id})
+        self.write({'payment_order_id': payment_order_id})
+        self.write({'state': 'finish'})

```

Comment:
Please do a single write with all values, it will do only one `UPDATE` in database instead of 3